core/src/main/java/org/jdbi/v3/ResultBearing.java

```diff
@@ -24,6 +31,8 @@
     @Override
     ResultIterator<ResultType> iterator();
 
+    ContainerFactoryRegistry getContainerFactoryRegistry();
```

Comment:
I did that because I wanted to reuse the code of collecting results to a container. Since we can use the `Collector.of` method, I think we could just implement the `collect` method  twice in `Query` and `GeneratedKeys` and didn't expose the internals.